src/msw/webview_ie.cpp

```diff
@@ -857,6 +857,67 @@ wxString wxWebViewIE::GetPageText() const
     }
 }
 
+wxWebViewIEEmulationLevel wxWebViewIE::GetEmulationLevel()
+{
+    wxRegKey key(wxRegKey::HKCU, _T("SOFTWARE\\Microsoft\\Internet Explorer\\Main\\FeatureControl\\FEATURE_BROWSER_EMULATION"));
+    if ( key.Exists() )
+    {
+        long val = 0;
+
+        if ( !key.QueryValue(wxGetFullModuleName().AfterLast('\\'), &val) )
+        {
+            wxLogWarning(_("Can not get level value of registry key \
+                SOFTWARE\\Microsoft\\Internet Explorer\\Main\\FeatureControl\\FEATURE_BROWSER_EMULATION"));
+            return wxWEBVIEWIE_EMULATION_LEVEL_EMPTY;
+        }
+
+        switch ( val )
+        {
+            case wxWEBVIEWIE_EMULATION_LEVEL_11:
+                return wxWEBVIEWIE_EMULATION_LEVEL_11;
+            case wxWEBVIEWIE_EMULATION_LEVEL_11_DEFAULT:
+                return wxWEBVIEWIE_EMULATION_LEVEL_11_DEFAULT;
+            case wxWEBVIEWIE_EMULATION_LEVEL_10:
+                return wxWEBVIEWIE_EMULATION_LEVEL_10;
+            case wxWEBVIEWIE_EMULATION_LEVEL_10_DEFAULT:
+                return wxWEBVIEWIE_EMULATION_LEVEL_10_DEFAULT;
+            case wxWEBVIEWIE_EMULATION_LEVEL_9:
+                return wxWEBVIEWIE_EMULATION_LEVEL_9;
+            case wxWEBVIEWIE_EMULATION_LEVEL_9_DEFAULT:
+                return wxWEBVIEWIE_EMULATION_LEVEL_9_DEFAULT;
+            case wxWEBVIEWIE_EMULATION_LEVEL_8:
+                return wxWEBVIEWIE_EMULATION_LEVEL_8;
+            case wxWEBVIEWIE_EMULATION_LEVEL_8_DEFAULT:
+                return wxWEBVIEWIE_EMULATION_LEVEL_8_DEFAULT;
+            case wxWEBVIEWIE_EMULATION_LEVEL_7_DEFAULT:
+                return wxWEBVIEWIE_EMULATION_LEVEL_7_DEFAULT;
+            default:
+                return wxWEBVIEWIE_EMULATION_LEVEL_EMPTY;
```

Comment:
`wxFAIL_MSG()` is always _available_, but it is _inactive_ in production builds. This is an important difference.

As for `UNKNOWN`, I had mentioned it in [this comment](https://github.com/wxWidgets/wxWidgets/pull/538#discussion_r134260834) which is currently hidden because it's "outdated" (but please expand the "Show outdated" markers to see it and other hidden comments, they're still valid).

But after reading the entire PR, I think we need neither `UNKNOWN` nor `EMPTY` nor this enum at all, and unless there is some important benefit in keeping them I'd simplify the API drastically to basically put it into "everything works" mode by default and allow switching it back to compatibility mode if really necessary.